MADE/made.jl

```diff
@@ -0,0 +1,140 @@
+#=
+Implements Masked AutoEncoder for Density Estimation, by Germain et al. 2015
+Re-implementation by Andrej Karpathy based on https://arxiv.org/abs/1502.03509
+Re-Re-implementation by Tejan Karmali using Flux.jl ;)
+=#
+
+using Flux
+using Flux: glorot_uniform, @epochs
+using NPZ
+
+# ------------------------------------------------------------------------------
+
+add_dims_r(a) = reshape(a, size(a)..., 1)
+add_dims_l(a) = reshape(a, 1, size(a)...)
+
+# ------------------------------------------------------------------------------
+
+struct MaskedDense{F,S,T,M}
+  # same as Linear except has a configurable mask on the weights
+  W::S
+  b::T
+  mask::M
+  σ::F
+end
+
+function MaskedDense(in::Integer, out::Integer; σ = identity)
+  return MaskedDense(param(glorot_uniform(out, in)), param(zeros(out)), ones(out, in), σ)
```

Comment:
This could probably be modified to be more type-generic? Would be nice to be able to support GPU and/or Float32

Reply:
Could you please explain a bit more on that? Because `in` and `out` have to be integers, `MaskedDense` has been modeled after `Dense`.